Zend/zend_compile.c

```diff
@@ -6762,6 +6767,10 @@ void zend_compile_params(zend_ast *ast, zend_ast *return_type_ast, uint32_t fall
 					ZSTR_VAL(scope->name), ZSTR_VAL(name), ZSTR_VAL(str));
 			}
 
+			if (!(property_flags & ZEND_ACC_READONLY) && (scope->ce_flags & ZEND_ACC_READONLY_CLASS)) {
+				property_flags |= ZEND_ACC_READONLY;
```

Comment:
This way, I don't have to add additional runtime checks whether the property is in a readonly class. On the other hand, this behavior is inconsistent with how we treat final methods and final classes.